### ISSUE TITLE
Local variable defined in try can be used outside of it

### ISSUE THREAD

**Issue by teggotic** [jemc/mare#55]
![image](https://user-images.githubusercontent.com/16068886/89123549-f2d08e00-d4d8-11ea-90c8-5ca59f577872.png)

![image](https://user-images.githubusercontent.com/16068886/89123486-6756fd00-d4d8-11ea-89cc-c75c9e736ce7.png)

**Comment by teggotic** [comment]
It may be caused by the fact that this try block is `maybe_error` and not `always_error`. If I try to compile this 
![image](https://user-images.githubusercontent.com/16068886/89123611-493dcc80-d4d9-11ea-89c5-95d290b63c1d.png), I get such error 
![image](https://user-images.githubusercontent.com/16068886/89123626-61155080-d4d9-11ea-99f8-d901ddbba2e5.png)

**Comment by jemc** [comment]
The latter issue is different from the former, I think.

For the latter issue, we need to detect unreachable code so we can skip generating code for it. We probably need to add that unreachable code detection to the jumps pass, or maybe a new pass that depends on the jumps pass. We will need that kind of detection for early return as well.

**Comment by jemc** [comment]
I will investigate the former issue

**Comment by jemc** [comment]
For the former issue, we need to improve the refer pass by noticing that a local variable that occurs after a "maybe error" is not defined in all cases, so is invalid to use, issuing a compile error for that in the refer pass

**Comment by jemc** [comment]
Better control flow analysis will help this bug, so I'm marking this ticket as blocked by this other one: https://github.com/jemc/mare/issues/75

**Comment by jemc** [comment]
This was fixed as part of the work done for #75, which is now finished.

The example above now gives an error that looks like this:
```
This local variable isn't guaranteed to have a value yet:
    env.out.print(local_var)
                  ^~~~~~~~~

- this assignment is not guaranteed to precede that usage:
      local_var = arr[10]!
      ^~~~~~~~~
```
